find/main/routes.py

```diff
@@ -175,6 +178,49 @@ def retrieve_download(filename: str):
         return render_template("find/main/retrieve-download.html", context=context, form=form)
 
 
+@bp.route("/retrieve-spreadsheet/<fund_code>/<submission_id>", methods=["GET", "POST"])
+@login_required(return_app=SupportedApp.POST_AWARD_FRONTEND)
+@check_internal_user
+def retrieve_spreadsheet(fund_code: str, submission_id: str):
+    object_name = f"{fund_code}/{submission_id}"
+
+    try:
+        file_metadata = get_file_metadata(Config.AWS_S3_BUCKET_SUCCESSFUL_FILES, object_name, True)
+    except FileNotFoundError:
+        if request.method == "POST":
+            return redirect(
+                url_for(
+                    ".retrieve_spreadsheet",
+                    fund_code=fund_code,
+                    submission_id=submission_id,
+                )
+            )
+
+        return render_template("find/main/spreadsheet-not-found.html")
+
+    filename = f"{fund_code}---{submission_id}.xlsx"
```

Comment:
suggestion: Should/could we also include the LA name to potentially be more user-friendly? Wonder if the date is useful as well? Maybe `{date} - {fund_code} - {submission_id} - {org_name}` - ie `2024-08-29 - TF - S-R05-1 - Bolton Council`?

nit: not sure we need three dashes in the filename, that feels a bit unusual to me.

Reply:
Changing it to:
`{date}-{fund-code}-{submission.id}` 
This is easy because I have this data already.

To get `submission.submission_id` and `org_name`. I'd have to go to the db which is a bit ugh..

Reply:
The org name - to me - feels a very useful thing for the fund team to have in the file name. I think we should include it?

Reply:
2024-07-08-pf-bolton-council-jan2024-mar2024.xlsx

This seems to have all the info they would need. Ideally, we'd want to find out from users what would be the most useful file naming convention for them, but since we're not doing any research on Submit MVP at the moment, I guess this isn't really possible.